### PR TITLE
doc: initial version of dependencies section

### PR DESCRIPTION
**Pull request by mhdawson**
Signed-off-by: Michael Dawson <mdawson@devrus.com>

### CODE REVIEW

**Comment by mhdawson** [comment]
In the last Ref arch meeting we agreed this should be pulled out as a separate section from the secure development process section.

**Review by lholmquist — APPROVED** [review]

File: docs/development/dependencies.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+---
+sidebar_position: 7
+---
+
```
Comment (mhdawson) [inline docs/development/dependencies.md:4]:
```suggestion
```

File: docs/development/dependencies.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+* Limit dependencies to the minimum required
+  * prefer dependencies with limited or no dependencies of their own
+  * avoid one-liners, is using the modules really saving you time 
+    development/maintenance time versus incorporating your own code?
+* Prefer dependencies that already have a reasonable level of usage 
+  based on github star count and npm usage count. Compare usage
+  to other possible options.
+* Prefer dependencies with an active, open, and inclusive community. 
+  Note that in some cases a dependency may be "done" but open and 
+  inclusive is still immportant in those cases. Some
```
Comment (helio-frota) [inline docs/development/dependencies.md:32]:
```suggestion
  inclusive is still important in those cases. Some
```

File: docs/development/dependencies.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  indicators to consider include:
+  * Number of active contributors since inception
+  * If you needed to contribute a fix would it be accepted?
+  * How long does it take for CVEs to get addressed?
+  * Do you have to move up to a new Major release to get security fixes?
+  * What is the cadence of releases?
+  * Is the project part of a larger project or Foundation?
+* Prefer depdencies aligned with the Node.js project releases
```
Comment (helio-frota) [inline docs/development/dependencies.md:40]:
```suggestion
* Prefer dependencies aligned with the Node.js project releases
```

File: docs/development/dependencies.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  * Does it support the current Node.js LTS releases
+  * Is the dependency tested by the Node.js project's
+    [Canary in the Gold Mine](https://github.com/nodejs/citgm)?
+  * If the depenency is a native module does it use Node-API (use
```
Comment (helio-frota) [inline docs/development/dependencies.md:44]:
```suggestion
  * If the dependency is a native module does it use Node-API (use
```

File: docs/development/dependencies.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    of Node-API means that packages can work across Node.js major versions)
+* Only use depenencies with licences acceptable to your organization
```
Comment (helio-frota) [inline docs/development/dependencies.md:46]:
```suggestion
* Only use dependencies with licences acceptable to your organization
```

File: docs/development/dependencies.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+* When choosing to use dependencies with a lower level of use in the 
+  ecosystem doing a code review to look for red flags can be useful.
+
+When evaluating dependencies, the teams typically evaluate the top level
+dependencies versus the full tree **EXCEPT** for the following in which
+we evalute the full tree:
```
Comment (helio-frota) [inline docs/development/dependencies.md:52]:
```suggestion
we evaluate the full tree:
```

**Review by helio-frota — APPROVED** [review]

File: docs/development/dependencies.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+* Only use dependencies with licences acceptable to your organization
+* When choosing to use dependencies with a lower level of use in the 
+  ecosystem doing a code review to look for red flags can be useful.
+
+When evaluating dependencies, the teams typically evaluate the top level
+dependencies versus the full tree **EXCEPT** for the following in which
+we evaluate the full tree:
+* Licence Checks
+* Security vulnerabilities (CVEs)
+* Total sub-dependencies
```
Comment (mhdawson) [inline docs/development/dependencies.md:55]:
```suggestion
Total number of sub-dependencies
```

**Comment by helio-frota** [comment]
probably need to add these 2 packages in npcheck.json?

"tslib@2.3.1" ----  package which is under the non-acceptable license "0BSD".
"caniuse-lite@1.0.30001282" ---- package which is under the non-acceptable license "CC-BY-4.0".

**Pull request merged**